tests/test_publish.py

```diff
@@ -1289,3 +1290,44 @@ def test_update_database_without_media(tmpdir, persistent_repository):
             audeer.path(build_root, attachment_file),
             audeer.path(db_load.root, attachment_file),
         )
+
+
+def test_unstability_of_media_files_order_during_publication():
+    with tempfile.TemporaryDirectory() as tmp:
+        name = 'db'
+        version = '1.0.0'
+
+        host = audeer.mkdir(audeer.path(tmp, 'host'))
+        db_root = audeer.mkdir(audeer.path(tmp, name))
+        cache_root = audeer.mkdir(audeer.path(tmp, 'cache'))
+
+        sampling_rate = 8000
+        duration = 1
+        signal = np.zeros((1, duration * sampling_rate))
+        audiofile.write(
+            audeer.path(db_root, 'f0.wav'), signal, sampling_rate
+        )
+        audiofile.write(
+            audeer.path(db_root, 'f1.wav'), signal, sampling_rate
+        )
+
+        db = audformat.Database(name)
+        index = audformat.filewise_index(['f0.wav', 'f1.wav'])
+        db['table'] = audformat.Table(index)
+        db['table']['column'] = audformat.Column()
+        db['table']['column'].set(['label0', 'label1'])
+        db.save(db_root)
+
+        repository = audb.Repository(
+            name='repo',
+            host=host,
+            backend='file-system',
+        )
+        audb.config.REPOSITORIES = [repository]
+        audb.config.CACHE_ROOT = cache_root
+
+        audb.publish(db_root, version, repository)
+
+        deps = audb.dependencies(name, version=version)
+
+        assert deps.files == sorted(deps.files)
```

Comment:
Actually, can't we simply move this line into `test_publish`? I don't really see the need to introduce an extra test function.

Reply:
Instead of sorting them alphabetically, maybe it would also make sense to first list all table files alphabetically sorted, then all attachment files, and then all media files.
But if this makes it more complicated and slower I would also be fine with sorting all files.

Reply:
Good point. To me it's actually not really clear why we need the sorting at all. But if we want that the dependency table always looks the same, we could also simply sort the table by index in the end.

Reply:
> But if this makes it more complicated and slower I would also be fine with sorting all files.

As far as I see we only sort the media files at the moment, which is what is returned by `deps.files` I guess?

Reply:
The current behavior when querying the dependencies is that the media files are being altered, however the order of the rest of metadata files is always the same. This is because we use `Set` structures in `_find_media()` and `_put_media()` but not in the methods that add/append the metadata to the dependencies table.

Reply:
> we could also simply sort the table by index in the end.

Do you mean from the user-end so we don't need to change anything in audb?

Reply:
> This is because we use Set structures in _find_media() and _put_media() but not in the methods that add/append the metadata to the dependencies table.

Ok, then the proposed fix seems fine to me and I don't think we need to do anymore sorting.